service/handlers.go

```diff
@@ -0,0 +1,149 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type WebhookRequest map[string]interface{}
+
+func Construct(w http.ResponseWriter, r *http.Request) {
+	var webhookRequestData WebhookRequest
+	log.Infof("Construct Payload")
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Construct failed with error: %v", err)
+		return
+	}
+	json.Unmarshal(bytes, &webhookRequestData)
+	_, err = ConstructPayload(webhookRequestData)
+}
+
+// ConstructPayload accepts and sends request to driver
+func ConstructPayload(parameters map[string]interface{}) (map[string]interface{}, error) {
+	cattleUrl := "http://localhost:8080/v2-beta"
+	webhookType := parameters["driver_type"]
+	switch webhookType {
+	case "service":
+		serviceId := parameters["service_id"].(string)
+		serviceAccountId := parameters["service_account_id"].(string)
+		url := fmt.Sprintf("%s/projects/%s/services/%s", cattleUrl, serviceAccountId, serviceId)
+		parameters["url"] = url
+		return parameters, nil
+	default:
+		err := fmt.Errorf("Driver not recognized")
+		return nil, err
+	}
+	return nil, nil
+}
+
+func Execute(w http.ResponseWriter, r *http.Request) {
+	var payload WebhookRequest
+	log.Infof("Execute")
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Construct failed with error: %v", err)
+		return
+	}
+	json.Unmarshal(bytes, &payload)
+	Process(payload)
+}
+
+//Process executes the driver
+func Process(payload map[string]interface{}) error {
```

Comment:
Good start, but we want this to have a cleaner driver architecture. To accomplish this, we need a few changes:
1. You need to define a driver interface that has two methods. It would look something like this:
  ```
  type WebhookDriver interface {
    ConstructPayload(input map[string]interface{}) (map[string] interface{}, error)
    Execute(payload map[string]interface{}) (int, map[string] interface{}, error) // int is the http response code to return to the user, map should be marshalled to json and returned as the body
````
2. create a map of keys to drivers and create a RegisterWebhookHook drivers method that instantiates each driver and puts it into the map
3. refactor this method to just look up the driver in the map
4. Move the drivers into a separate driver package.